hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/PipelineReportHandler.java

```diff
@@ -102,12 +106,24 @@ private void processPipelineReport(PipelineReport report, DatanodeDetails dn)
       return;
     }
 
+    if (report.hasLeaderID()) {
+      Map<UUID, ByteString> ids =
+          reportedLeadersForPipeline.computeIfAbsent(pipelineID,
+              k -> new HashMap<>());
+      ids.put(dn.getUuid(), report.getLeaderID());
+    }
+
     if (pipeline.getPipelineState() == Pipeline.PipelineState.ALLOCATED) {
```

Comment:
This will not cover OPEN pipelines where there is a leader election. We need to update SCM's internal pipeline's leaderId to the new leader for those leader elections as well.